core/network/impl/rating_repository_impl.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "network/impl/rating_repository_impl.hpp"
+
+#include <boost/assert.hpp>
+
+namespace kagome::network {
+
+  PeerRatingRepositoryImpl::PeerRatingRepositoryImpl(
+      std::shared_ptr<libp2p::basic::Scheduler> scheduler)
+      : scheduler_{std::move(scheduler)} {
+    BOOST_ASSERT(scheduler_);
+  }
+
+  PeerScore PeerRatingRepositoryImpl::rating(
+      const PeerRatingRepository::PeerId &peer_id) const {
+    return rating_table_.count(peer_id) == 0 ? 0 : rating_table_.at(peer_id);
+  }
+
+  PeerScore PeerRatingRepositoryImpl::upvote(
+      const PeerRatingRepository::PeerId &peer_id) {
+    ensurePeerPresence(peer_id);
+    return ++rating_table_[peer_id];
```

Comment:
[operator[]](https://en.cppreference.com/w/cpp/container/unordered_map/operator_at) inserts default value if key is not present.  
Can `ensurePeerPresence` be removed?

Reply:
No. The part regarding [] is true. Avoiding ensure call will result in default initialization of int value field which would be compiler-defined (generally UB), thus it may not be the same on different platforms/compilers/build types.

For example, about 5 years ago MSVC intentionally did different initialization for variables, which were not initialized clearly, with different byte patters in Release and Debug builds.